core/src/main/java/io/github/belgif/rest/guide/validator/core/ApiFunctions.java

```diff
@@ -50,20 +50,30 @@ public static boolean existsPathWithPathParamAfter(String pathString, Parser.Par
      * @param schema Schema that has to be validated
      */
     public static boolean schemaMeetsCondition(Schema schema, Parser.ParserResult result, Predicate<SchemaDefinition> condition) {
+        return schemaMeetsCondition(schema, result, condition, new HashSet<>());
+    }
+
+    private static boolean schemaMeetsCondition(Schema schema, Parser.ParserResult result, Predicate<SchemaDefinition> condition, Set<Schema> checkedSubSchemas) {
+        if (checkedSubSchemas.contains(schema)) {
```

Comment:
logic is good, but name "checkedSubSchemas" is a bit confusing:

checkedSubSchemas -> parentSchemas?
branchedCheckedSubSchemas -> parentSchemasOfChildren